modnet/models.py

```diff
@@ -370,127 +397,113 @@ def fit_preset(
-                    train_data = data
-
-                self.model = MODNetModel(
-                    self.targets,
-                    self.weights,
-                    num_neurons=params["num_neurons"],
-                    n_feat=n_feat,
-                    act=params["act"],
-                    num_classes=self.num_classes
-                ).model
-                self.n_feat = n_feat
-                self.fit(
-                    train_data,
-                    lr=params["lr"],
-                    epochs=params["epochs"],
-                    batch_size=params["batch_size"],
-                    loss=params["loss"],
-                    callbacks=callbacks,
-                    verbose=verbose,
-                    **val_params
-                )
-
-                learning_curve = self.model.history.history["val_loss"]
-
-                if any(isinstance(cb, keras.callbacks.EarlyStopping) for cb in callbacks):
-                    val_loss = np.min(learning_curve)
-                    mean_val_loss = np.mean(learning_curve[-20:])
-                    LOG.info(f"Taking loss {val_loss:3.3f} from EarlyStopping vs mean {mean_val_loss:3.3f}")
-                else:
-                    val_loss = np.mean(learning_curve[-20:])
-                    LOG.info(f"Loss from curve {val_loss:3.3f}")
-
-                nested_val_losses.append(val_loss)
-                nested_learning_curves.append(learning_curve)
-                nested_models.append(self.model)
-
-            val_loss = np.mean(nested_val_losses)
-
-            if val_loss < min(val_losses):
-                best_model = self.model
-                best_n_feat = n_feat
-                best_scaler = self._scaler
-
-            val_losses[i] = val_loss
-
-            models.append(nested_models)
-            learning_curves.append(nested_learning_curves)
-            LOG.info("Validation loss for preset #{}, {}: {:.3f}".format(i+1, params["loss"], val_loss))
-
-        best_preset_idx = val_losses.argmin()
+                train_data, val_data = data.split((train, val))
+                val_params["val_data"] = val_data
+
+                tasks += [{'train_data' : train_data,
+                   'targets' : self.targets,
+                   'weights' : self.weights,
+                   'num_classes' : self.num_classes,
+                   'n_feat' : n_feat,
+                   'num_neurons' : params["num_neurons"],
+                   'lr' : params["lr"],
+                   'batch_size' : params["batch_size"],
+                   'epochs' : params["epochs"],
+                   'loss' : params["loss"],
+                   'act' : params["act"],
+                   'out_act': self.out_act,
+                   'callbacks' : callbacks,
+                   'preset_id' : i,
+                   'fold_id' : ind,
+                   'verbose' : verbose,
+                   **val_params,
+                }]
+
+        val_losses = np.zeros((len(presets),n_splits))
```

Comment:
Recommend keeping this as it was, so it is obvious when something has gone wrong:

```suggestion
        val_losses = 1e20 * np.ones((len(presets),))
```